crates/ty_python_semantic/src/types/typed_dict.rs

```diff
@@ -76,6 +78,174 @@ impl<'db> TypedDictType<'db> {
             ),
         }
     }
+
+    // Subtyping between `TypedDict`s follows the algorithm described at:
+    // https://typing.python.org/en/latest/spec/typeddict.html#subtyping-between-typeddict-types
+    pub(super) fn has_relation_to_impl(
```

Comment:
This is a really clear and well-documented implementation of some fairly hairy logic!